dhis-2/dhis-support/dhis-support-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterManager.java

```diff
@@ -0,0 +1,167 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.fieldfiltering;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.fieldfiltering.transformers.IsEmptyFieldTransformer;
+import org.hisp.dhis.fieldfiltering.transformers.IsNotEmptyFieldTransformer;
+import org.hisp.dhis.fieldfiltering.transformers.PluckFieldTransformer;
+import org.hisp.dhis.fieldfiltering.transformers.RenameFieldTransformer;
+import org.hisp.dhis.fieldfiltering.transformers.SizeFieldTransformer;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.OrderComparator;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+
+/**
+ * @author Morten Olav Hansen
+ */
+@Component
+@RequiredArgsConstructor
+public class FieldFilterManager
+{
+    @Qualifier( "jsonMapper" )
+    private final ObjectMapper jsonMapper;
+
+    public List<ObjectNode> toObjectNode( FieldFilterParams<?> params )
+    {
+        List<FieldPath> fieldPaths = FieldFilterParser.parse( params.getFilters() );
+
+        SimpleFilterProvider filterProvider = getSimpleFilterProvider( fieldPaths );
+        ObjectMapper objectMapper = jsonMapper.setFilterProvider( filterProvider );
+        Map<String, List<FieldTransformer>> fieldTransformers = getTransformers( fieldPaths );
+
+        List<ObjectNode> objectNodes = new ArrayList<>();
+
+        for ( Object object : params.getObjects() )
+        {
+            ObjectNode objectNode = objectMapper.valueToTree( object );
+            applyTransformers( objectNode, null, "", fieldTransformers );
+
+            objectNodes.add( objectNode );
+        }
+
+        return objectNodes;
+    }
+
+    private void applyTransformers( JsonNode node, JsonNode parent, String path,
+        Map<String, List<FieldTransformer>> fieldTransformers )
+    {
+        if ( parent != null && !parent.isArray() && !path.isEmpty() )
+        {
+            List<FieldTransformer> transformers = fieldTransformers.get( path.substring( 1 ) );
+
+            if ( transformers != null )
+            {
+                transformers.sort( OrderComparator.INSTANCE );
```

Comment:
Wouldn't it make more sense to sort the transformers in the `getTransformers` once and to assume from there on that whatever list of transformers you have that this is the one order that should be applied?